torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -80,19 +80,15 @@ Tensor FlattenImpl::forward(const Tensor& input) {
 
 // ============================================================================
 
-UnflattenImpl::UnflattenImpl(const UnflattenOptions& options_) : options(options_) {
-  reset();
-}
+UnflattenImpl::UnflattenImpl(const UnflattenOptions& options_) : options(options_) {}
```

Comment:
it seems clang-tidy follow morden c++ convention.
You need either add a move constructor:
UnflattenImpl::UnflattenImpl(UnflattenOption&& options_): options(std::move(options_){}

or replace current const ref constructor with the following one:
UnflattenImpl::UnflattenImpl(UnflattenOptions options_) : options(std::move(options_) {}

UnflattenImpl doesn't create a default move constructor, I doubt this is due to UnflattenOptions contains variant which compiler is not sure whether it is movable.